### DISCUSSION TITLE
Initial camera position

### DISCUSSION THREAD

**Discussion by fabricDev** [mkkellogg/GaussianSplats3D#344] [Q&A] [answered]
Is there a way to calculate the "initialCameraPosition" and "initialCameraLookAt" dynamically for any given splat file before rendering ? We need the file to be rendered with correct position when its opened for the first time.

**Comment by rowild** [comment]
What I did was opening the scene, hitting "I" (for the info window to pop up), moving around the camera until I found the position I liked, and eventually copied the values shown in the info window.

**Reply by fabricDev** [reply]
Thanks for the reply. My requirement is to have splat file rendered with correct position when file is opened for the first time itself. Need a generic logic that would calculate position dynamically.

**Reply by rowild** [reply]
And how s/would the script logic know what the "proper" position should be?

**Reply by fabricDev** [reply]
If the logic can evaluate the centroid using the splat file buffer and the screen size 
that would suffice. For some of the files I am using,  its quite difficult for me to move and get into the position I like after it renders.

**Reply by rowild** [reply]
I understand the moving-to and positioning problem. I think, though, my use cases are a whole lot smaller than yours are, so my contributions here are very likely not to help at all. Nevertheless let me just drop SuperSplat as a final comment – have you heard of it? I use it to clean up and reposition my splats the way I need them. And they open fine in GaussianSplats3D. (But then again my splats are small, usually flying around a castle; and I do get rid of all those huge amounts of landscapes and surroundings, so the final object is indeed rather small... I have a hunch your requirements and demands are way bigger...)

**Comment by mkkellogg** [comment, marked as answer]
@fabricDev If you need the center of the splat scene, you could conceivably use `SplatMesh.computeBoundingBox(applySceneTransforms, sceneIndex)` to first calculate the scene bounds, and then just use the its center. I think you'll most likely want to pass true for the `applySceneTransforms` parameter, although if that doesn't work you can always try passing false :)
